src/NzbDrone.Core/Authentication/UserService.cs

```diff
@@ -13,12 +14,22 @@ namespace NzbDrone.Core.Authentication
 {
     public interface IUserService
     {
-        User Add(string username, string password);
+        User Add(string username, string password, UserRole role);
         User Update(User user);
         User Upsert(string username, string password);
         User FindUser();
         User FindUser(string username, string password);
         User FindUser(Guid identifier);
+        User UpdateByModel(User updatedUser);
+        public User FindUser(int id);
+
+        void Delete(int id);
+
+        public User FindUserFromApiKey(string apiKey);
```

Comment:
The interface doesn't need to define that they are public and should be removed to match the existing methods.

Reply:
Hi, the public accessors are needed because they are used by other classes. For example in ApiKeyAuthenticationHandler.cs

`var user = _userService.FindUserFromApiKey(providedApiKey);`

I'm getting a compile error If I remove this remove these public addresses. Unless you mean something else I am missing  @markus101.

Reply:
Can ApiKeyAuthenticationHandler rely on IUserService instead of the concrete type?